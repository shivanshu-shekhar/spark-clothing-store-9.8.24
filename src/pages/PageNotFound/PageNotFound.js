import React from "react";
import pagenotfoundlogo from "../../assets/images/page-not-found.jpg";
import { NavLink } from "react-router-dom";
import { Helmet } from "react-helmet";

const PageNotFound = () => {
  return (
    <div className="mt-5">
      {/* page title in the browser */}
      <Helmet>
        <title>ERROR 404 PAGE</title>
      </Helmet>
      {/* heading of the page not found */}
      <h1 className="text-center">
        Page not Found Error 404
        <img className="mt-3" src={pagenotfoundlogo} alt="page not found logo"></img>
        <p>Navigate to Home Page</p>
        {/* navigate to the homepage button */}
        <NavLink to="/">
          <button className="border rounded-pill" data-testid="goHome">
            Home
          </button>
        </NavLink>
      </h1>
    </div>
  );
};

export default PageNotFound;
