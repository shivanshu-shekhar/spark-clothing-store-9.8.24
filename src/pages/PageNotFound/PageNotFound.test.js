import { render, screen } from "@testing-library/react";
import { HashRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import PageNotFound from "./PageNotFound";

//Go to home button testing
it("Is go to home button working", () => {
  render(
    <HashRouter>
      <PageNotFound />
    </HashRouter>
  );
  const goHome = screen.getByTestId("goHome");
  expect(goHome).toHaveTextContent("Home");
});

test("renders correctly", () => {
  const tree = renderer
    .create(
      <HashRouter>
        <PageNotFound />
      </HashRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
