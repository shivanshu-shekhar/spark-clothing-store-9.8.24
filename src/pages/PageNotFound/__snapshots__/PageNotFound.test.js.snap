// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mt-5"
>
  <h1
    className="text-center"
  >
    Page not Found Error 404
    <img
      alt="page not found logo"
      className="mt-3"
      src="page-not-found.jpg"
    />
    <p>
      Navigate to Home Page
    </p>
    <a
      aria-current="page"
      className="active"
      href="#/"
      onClick={[Function]}
    >
      <button
        className="border rounded-pill"
        data-testid="goHome"
      >
        Home
      </button>
    </a>
  </h1>
</div>
`;
